src/routes/mods.rs

```diff
@@ -87,6 +132,289 @@ fn convert_mod(data: database::models::mod_item::QueryMod) -> models::mods::Mod
+        .await
+        .map_err(|e| ApiError::DatabaseError(e.into()))?;
+
+    if let Some(mod_item) = result {
+        let is_moderator = user.role == Role::Moderator || user.role == Role::Admin;
+        let team_member = database::models::TeamMember::get_from_user_id(
+            mod_item.inner.team_id,
+            user.id.into(),
+            &**pool,
+        )
+        .await?;
+        let permissions;
+
+        if let Some(member) = team_member {
+            permissions = Some(member.permissions)
+        } else if is_moderator {
+            permissions = Some(Permissions::ALL)
+        } else {
+            permissions = None
+        }
+
+        if let Some(perms) = permissions {
+            let mut transaction = pool
+                .begin()
+                .await
+                .map_err(|e| ApiError::DatabaseError(e.into()))?;
+
+            if let Some(title) = &new_mod.title {
+                if !perms.contains(Permissions::EDIT_DETAILS) {
+                    return Err(ApiError::CustomAuthenticationError(
+                        "You do not have the permissions to edit the title of this mod!"
+                            .to_string(),
+                    ));
+                }
+
+                sqlx::query!(
+                    "
+                    UPDATE mods
+                    SET title = $1
+                    WHERE (id = $2)
+                    ",
+                    title,
+                    id as database::models::ids::ModId,
+                )
+                .execute(&mut *transaction)
+                .await
+                .map_err(|e| ApiError::DatabaseError(e.into()))?;
+            }
+
+            if let Some(description) = &new_mod.description {
+                if !perms.contains(Permissions::EDIT_DETAILS) {
+                    return Err(ApiError::CustomAuthenticationError(
+                        "You do not have the permissions to edit the description of this mod!"
+                            .to_string(),
+                    ));
+                }
+
+                sqlx::query!(
+                    "
+                    UPDATE mods
+                    SET description = $1
+                    WHERE (id = $2)
+                    ",
+                    description,
+                    id as database::models::ids::ModId,
+                )
+                .execute(&mut *transaction)
+                .await
+                .map_err(|e| ApiError::DatabaseError(e.into()))?;
+            }
+
+            if let Some(status) = &new_mod.status {
+                if !perms.contains(Permissions::EDIT_DETAILS) {
+                    return Err(ApiError::CustomAuthenticationError(
+                        "You do not have the permissions to edit the status of this mod!"
+                            .to_string(),
+                    ));
+                }
+
+                if status == &ModStatus::Rejected || status == &ModStatus::Approved {
```

Comment:
Users shouldn't be able to modify mod's statuses *at all*, in any cases.  This *will* end up having bugs that allow users to approve their own mods or get around the approval system.

Reply:
Yes they should. If a user want's to make a mod draft to processing, they should be able to do so. (ETC)